app/controllers/repositories_controller.rb

```diff
@@ -2,7 +2,7 @@
 
 class RepositoriesController < ApplicationController
   before_action :authenticate_user!, except: [:show, :state, :state_with_date]
-  before_action :project_owner?, only: [:new, :create]
+  before_action :project_owner?, only: [:new, :create], if: "!params[:project_id].nil?"
```

Comment:
Can't you use `proc { code }` or a method here, instead of a string? It's always good to make sure code looks like what it is, instead of being inside strings :)